log/enabled.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package log // import "go.opentelemetry.io/otel/log"
+
+// EnabledParameters represents payload for [Logger]'s Enabled method.
+type EnabledParameters struct {
+	severity Severity
+}
+
+// Severity returns the [Severity] level.
+func (r *EnabledParameters) Severity() Severity {
+	return r.severity
+}
+
+// SetSeverity sets the [Severity] level.
+func (r *EnabledParameters) SetSeverity(level Severity) {
+	r.severity = level
+}
```

Comment:
Why have getters and setters here?

```go
type EnabledParameters struct {
	Severity Severity
}
```

This means a user would be able to do `logger.Enabled(ctx, EnabledParameters{Severity: SeverityInfo})` instead of:

```go
p := new(EnabledParameters)
p.SetSeverity(SeverityInfo)
logger.Enabled(ctx, p)
```

Reply:
Additionally using fields directly matches the `SamplingParameters` this is modeled after.